avm/src/vm/opcode.rs

```diff
@@ -2758,48 +2758,77 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
+      let mut out_headers = Vec::new();
+      for i in 0..out_headers.len() {
+        let header = hand_mem.read_from_fractal(&headers, i).0;
+        let key = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&header, 0).0);
+        let val = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&header, 1).0);
+        out_headers.push((key, val));
       }
-      hand_mem
-    })
-  });
-
-  async fn __httppost(uri: String, payload: String) -> Result<String, Box<dyn std::error::Error>> {
-    let url = TryInto::<Uri>::try_into(uri)?;
-    let request = Request::post(url).body(Body::from(payload))?;
-    let client = Client::builder().build::<_, Body>(HttpsConnector::new());
-    let mut response = client.request(request).await?;
-    let body = hyper::body::to_bytes(response.body_mut()).await?;
-    let res = String::from_utf8(body.to_vec())?;
-    Ok(res)
-  }
-  io!(httppost => fn(args, mut hand_mem) {
-    Box::pin(async move {
-      let uri = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0]));
-      let payload = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[1]));
-      let res = __httppost(uri, payload).await;
+      let body = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&req, 3).0);
+      let out_body = if body.len() > 0 { Some(body) /* once told me... */ } else { None };
+      let res = __httpreq(method, url, out_headers, out_body).await;
       hand_mem.init_fractal(args[2]);
       hand_mem.push_fixed(args[2], if res.is_ok() { 1i64 } else { 0i64 });
       match res {
-        Ok(res) => hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal(&res)),
-        Err(ee) => hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal(format!("{}", ee).as_str())),
+        Ok(mut res) => {
+          let mut res_hm = HandlerMemory::new(None, 3);
+          res_hm.init_fractal(0);
+          res_hm.push_fixed(0, res.status().as_u16() as i64);
+          let headers = res.headers();
+          let mut headers_hm = HandlerMemory::new(None, headers.len() as i64);
+          let mut i = 0;
+          for (key, val) in headers.iter() {
+            let key_str = key.as_str();
+            let val_str = val.to_str().unwrap();
+            headers_hm.init_fractal(i);
+            headers_hm.push_fractal(i, HandlerMemory::str_to_fractal(key_str));
+            headers_hm.push_fractal(i, HandlerMemory::str_to_fractal(val_str));
+            i = i + 1;
+          }
+          HandlerMemory::transfer(&headers_hm, 0, &mut res_hm, CLOSURE_ARG_MEM_START);
+          res_hm.push_register(0, CLOSURE_ARG_MEM_START);
+          let body = hyper::body::to_bytes(res.body_mut()).await;
+          if body.is_ok() {
+            let body_str = String::from_utf8(body.unwrap().to_vec()).unwrap_or("".to_string());
+            res_hm.push_fractal(0, HandlerMemory::str_to_fractal(&body_str));
+          } else {
+            res_hm.push_fractal(0, HandlerMemory::str_to_fractal(&""));
```

Comment:
same as above - if the body is invalid shouldn't we let the user handle it?

Reply:
Same as above as well, I will, but the code is starting to look not great with all of these branches.